actions/upload_sessions.go

```diff
@@ -215,3 +220,31 @@ func (usr *UploadSessionResource) CreateBeta(c buffalo.Context) error {
 	}
 	return c.Render(200, r.JSON(res))
 }
+
+
+func (usr *UploadSessionResource) GetPaymentStatus(c buffalo.Context) error {
+	session := models.UploadSession{}
+	err := models.DB.Where("id = ?", c.Param("id")).First(&session)
+
+	if err != nil {
```

Comment:
Does pop return this as an err or would `session` be nil?

Reply:
We actually ran into json syntax errors when trying to render something to return.  

We need to render two types of errors I suppose:
-when something is actually wrong
-when nothing is wrong but there's just no session matching that ID (we were going to use 204 for this)  

@jfarago You could take a look at some of the other files in the actions folder, particularly for methods that are handlers for URLs that end with {id}.  There may already be an example of returning an error when no such session exists.

Reply:
we can hold off on rendering better errors for now. since we are the only client, we know how to handle it.

However when something is not found, will pop give you an error or an empty struct? If an empty struct, we should check `if session == nil || err != nil`

Reply:
I am not entirely sure if it gives you an empty struct in this case.  I think it may just give you a struct in which all fields have been assigned to default values, such as "" for strings and 0 for ints, etc.  

I often use .All and then check for len(arrayOfThings) == 0 for this reason, but another option would be to still use .First and check that session.ID == (whatever the default value for a uuid would be)